generators/pulsar/generator.mjs

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright 2013-2023 the original author or authors from the JHipster project.
+ *
+ * This file is part of the JHipster project, see https://www.jhipster.tech/
+ * for more information.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import _ from 'lodash';
+
+import BaseApplicationGenerator from '../base-application/index.mjs';
+import { GENERATOR_PULSAR, GENERATOR_BOOTSTRAP_APPLICATION_SERVER } from '../generator-list.mjs';
+import writePulsarFilesTask from './files.mjs';
+
+/**
+ * @typedef {import('../server/types.mjs').SpringBootApplication} SpringBootApplication
+ */
+/**
+ * @class
+ * @extends {BaseApplicationGenerator<SpringBootApplication>}
+ */
+export default class PulsarGenerator extends BaseApplicationGenerator {
+  async beforeQueue() {
+    await this.dependsOnJHipster(GENERATOR_BOOTSTRAP_APPLICATION_SERVER);
+    if (!this.fromBlueprint) {
+      await this.composeWithBlueprints(GENERATOR_PULSAR);
+    }
+  }
+
+  get writing() {
+    return {
+      writePulsarFilesTask,
+    };
+  }
+
+  get [BaseApplicationGenerator.WRITING]() {
+    return this.delegateTasksToBlueprint(() => this.writing);
+  }
+
+  get postWriting() {
+    return this.asPostWritingTaskGroup({
+      addTestSpringFactory({ source, application }) {
```

Comment:
Files customizations were lost in rebase, we should use api instead:
```suggestion
      customizeSpring({ source, application }) {
        source.addIntegrationTestAnnotation?.({ package: `${application.packageName}.config`, annotation: 'EmbeddedPulsar' });
```
Re-add logs?
https://github.com/jhipster/generator-jhipster/blob/32f8f609d2a028c9c6e4e5cee17397537a6db1ea/generators/kafka/generator.mts#L48-L51